### ISSUE TITLE
Run our own model using EMTk

### ISSUE THREAD

**Issue by bcdasilv** [SentimentAnalysisInSE/code_review_analysis#5]
Instead of using the pre-built EMTk model which was trained over StackOverflow discussion, we should train and test a new model based on the dataset we're using (code review comments from SentiCR).

Use the oracle.xlsx dataset to train a new model in EMTk using 10-fold cross validation.

Report the performance of this new model in a wikipage in our repo.

**Comment by steven19992017** [comment]
when attempting to train my own emotion model,  i ran into the follow errors:
![image](https://user-images.githubusercontent.com/33203413/64257714-60d2a500-cf58-11e9-8609-ceed052c4465.png)
Additionally, I don't really understand instructions on formatting the file for training the new model. 
id;label;text
…
22;NO;"""Excellent! This is exactly what I needed. Thanks!"""
23;YES;"""FEAR!!!!!!!!!!!"""
…
Also, I don't see how we could convert our polarity score from oracle.xlsx to emotions or "Yes" and "No" as mentioned on the example.

**Comment by steven19992017** [comment]
hours spent: 8 hours

**Comment by bcdasilv** [comment]
Thanks for the update.

First, it seems that you missed the '-p' parameter. See that issue they've reported: https://github.com/collab-uniba/EMTk/issues/3

Also, make sure a file 'sample.csv' is available. Is that the sample file provided by them? 
Are you trying to test-drive first with their sample file? See that section on their readme: 
_"Users can test-drive the emotion classification module by using the file /emotions_sample.csv, which contains only a handful of documents. Other more complex sample datasets are available at /emotions/java/DatasetSO/StackOverflowCSV."_

Regarding the file format, I don't know exactly what you don't understand. The columns ID and text are straightforward. They mean the unique id of the text/row and the text is the actual content to be classified. However, the column 'label' does seem confusing to me. Maybe they're referring to the emotion label, although YES and NO don't seem to be emotion labels. They could be sentiment polarity labels (NO - non-negative, YES - negative) which they used as an example. So, if that's the same confusion you've seen, please open their csv files and check if there are other labels than just YES and NO. Also, hopefully, after fixing the issues above you can test-drive the emotion module and see the results to better understand the label column. Does it make sense?

Please, post updates with further questions, comments, or solutions. Thanks!

**Comment by steven19992017** [comment]
For some reason running docker desktop crashed on windows. I will look into the issue
![image](https://user-images.githubusercontent.com/33203413/64469411-e5ddda00-d163-11e9-9b2a-4ebc331a08d3.png) ,,

**Comment by bcdasilv** [comment]
That sucks. Any updates, @steven19992017 ?

**Comment by steven19992017** [comment]
I spent some time reinstalling docker as well as pulling the emtk image. I explored the example training files for the emtk model but I still find the labels of "Yes" and "No" confusing. Maybe they are for indicating whether or not they are gold standard datas? Besides, I don't see how just giving them sentences without classifying the actual emotions would train our new model. Based on my understanding, to train a model we need to tell them the correct emotion but they never mentioned anything about specifying our own emotion inputs.

**Comment by steven19992017** [comment]
Time log: 8 hours

**Comment by bcdasilv** [comment]
You're correct. In order to train, we would need a column to tell the correct emotion. The code review dataset we have (oracle.xls file) was labeled based on two-level sentiments (negative and non-negative). Let's talk more about it in our next meeting.
